tests/exchange/test_binance.py

```diff
@@ -162,168 +171,369 @@ def test_stoploss_adjust_binance(mocker, default_conf, sl1, sl2, sl3, side):
-            [50000.0, 0.005, 50.0],   # lev = 200.0
-            [250000.0, 0.01, 1300.0],   # lev = 100.0
-            [1000000.0, 0.025, 16300.0],   # lev = 40.0
-            [5000000.0, 0.05, 141300.0],   # lev = 20.0
-            [20000000.0, 0.1, 1141300.0],   # lev = 10.0
-            [50000000.0, 0.125, 2391300.0],   # lev = 8.0
-            [100000000.0, 0.15, 4891300.0],   # lev = 6.666666666666667
-            [200000000.0, 0.25, 24891300.0],   # lev = 4.0
-            [300000000.0, 0.5, 99891300.0],   # lev = 2.0
-        ]
-    }
-    assert exchange.get_max_leverage(pair, stake_amount) == max_lev
+    assert exchange.get_max_leverage("BNB/USDT", 100.0) == 1.0
+
+    # Test Futures
+    default_conf['trading_mode'] = 'futures'
+    default_conf['margin_mode'] = 'isolated'
+    exchange = get_patched_exchange(mocker, default_conf, id="binance")
+
+    exchange._leverage_tiers = leverage_tiers
 
+    assert exchange.get_max_leverage("BNB/BUSD", 1.0) == 20.0
+    assert exchange.get_max_leverage("BNB/USDT", 100.0) == 75.0
+    assert exchange.get_max_leverage("BTC/USDT", 170.30) == 125.0
+    assert isclose(exchange.get_max_leverage("BNB/BUSD", 99999.9), 5.000005)
+    assert isclose(exchange.get_max_leverage("BNB/USDT", 1500), 33.333333333333333)
+    assert exchange.get_max_leverage("BTC/USDT", 300000000) == 2.0
+    assert exchange.get_max_leverage("BTC/USDT", 600000000) == 1.0  # Last tier
 
-def test_fill_leverage_brackets_binance(default_conf, mocker):
+    assert exchange.get_max_leverage("ETC/USDT", 200) == 1.0    # Pair not in leverage_tiers
+    assert exchange.get_max_leverage("BTC/USDT", 0.0) == 125.0  # No stake amount
+    with pytest.raises(
+        InvalidOrderException,
+        match=r'Amount 1000000000.01 too high for BTC/USDT'
+    ):
+        exchange.get_max_leverage("BTC/USDT", 1000000000.01)
+
+
+def test_fill_leverage_tiers_binance(default_conf, mocker):
     api_mock = MagicMock()
-    api_mock.load_leverage_brackets = MagicMock(return_value={
-        'ADA/BUSD': [[0.0, 0.025],
-                     [100000.0, 0.05],
-                     [500000.0, 0.1],
-                     [1000000.0, 0.15],
-                     [2000000.0, 0.25],
-                     [5000000.0, 0.5]],
-        'BTC/USDT': [[0.0, 0.004],
-                     [50000.0, 0.005],
-                     [250000.0, 0.01],
-                     [1000000.0, 0.025],
-                     [5000000.0, 0.05],
-                     [20000000.0, 0.1],
-                     [50000000.0, 0.125],
-                     [100000000.0, 0.15],
-                     [200000000.0, 0.25],
-                     [300000000.0, 0.5]],
-        "ZEC/USDT": [[0.0, 0.01],
-                     [5000.0, 0.025],
-                     [25000.0, 0.05],
-                     [100000.0, 0.1],
-                     [250000.0, 0.125],
-                     [1000000.0, 0.5]],
+    api_mock.fetch_leverage_tiers = MagicMock(return_value={
+        'ADA/BUSD': [
```

Comment:
Is there a substancial difference between this data and the data in conftest? (like - are there fields that are binance specific and we use)?

if it's only different pairs - i'd keep the data in one place (which is conftest - not individual test files).

Reply:
`conftest` only has the output of this method, here I'm testing if we get the expected output from a given input

Reply:
is that input exchange-specific to binance? I'd assume ccxt will return the same(ish) for all exchanges?

Reply:
The data in `info` is exchange specific to binance, and we use the variable `cum` in info